executor/adapter.go

```diff
@@ -1028,6 +1038,7 @@ func (a *ExecStmt) SummaryStmt(succ bool) {
 	memMax := stmtCtx.MemTracker.MaxConsumed()
 	diskMax := stmtCtx.DiskTracker.MaxConsumed()
 	sql := a.GetTextToLog()
+	hints := a.GetHints()
```

Comment:
There's no need to always extract hints every time, it may take some time.
You can just pass a function to `StmtExecInfo` and call it only at the first time, just like what `planDigestGen` does.

Reply:
OK!